src/components/Api.js

```diff
@@ -0,0 +1,71 @@
+export default class Api {
+  constructor({baseUrl, headers}) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+_checkResponse(res){
+  if (res.ok) {
+    return res.json();
+  }else{
+    return Promise.reject(`'Error:' ${res.status}`)
+  }
+}
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+      }).then(this._checkResponse);
```

Comment:
**Could be improved**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.